autoload/ctrlp/mrufiles.vim

```diff
@@ -66,10 +67,10 @@ fu! s:record(bufnr)
 endf
 
 fu! s:addtomrufs(fname)
-	let fn = fnamemodify(a:fname, ':p')
+	let fn = fnamemodify(a:fname, g:ctrlp_mruf_tilde_homedir ? ':p:~' : ':p')
 	let fn = exists('+ssl') ? tr(fn, '/', '\') : fn
 	if ( !empty({s:in}) && fn !~# {s:in} ) || ( !empty({s:ex}) && fn =~# {s:ex} )
-		\ || !empty(getbufvar('^'.fn.'$', '&bt')) || !filereadable(fn) | retu
+		\ || !empty(getbufvar('^'.fnamemodify(fn,':p').'$', '&bt')) || !filereadable(fnamemodify(fn,':p')) | retu
```

Comment:
I think you could keep the previous code from line 69, but store it in a different variable (like `abs_fn` instead of `fn`) so you don't need to call `fnamemodify` twice here.